Data/ByteString/FastBuilder/Internal.hs

```diff
@@ -715,26 +723,26 @@ growBuffer !bufRef !req = do
 -- the 'Queue'.
 flushQueue :: IO.Handle -> IORef Queue -> Ptr Word8 -> IO ()
 flushQueue !h !qRef !cur = do
-  Queue fptr start <- readIORef qRef
+  Queue fptr start total <- readIORef qRef
   let !end = cur `minusPtr` unsafeForeignPtrToPtr fptr
   when (end > start) $ do
     S.hPut h $ S.fromForeignPtr fptr start (end - start)
-    writeIORef qRef $ Queue fptr end
+    writeIORef qRef $ Queue fptr end $ total + end - start
 
--- | @switchQueue qRef minSize@ discards the old 'Queue' and sets up
+-- | @switchQueue qRef minSize adv@ discards the old 'Queue' and sets up
 -- a new empty 'Queue' of at least @minSize@ large. If the old 'Queue'
--- is large enough, it is re-used.
-switchQueue :: IORef Queue -> Int -> BuildM ()
-switchQueue !qRef !minSize = do
+-- is large enough, it is re-used. Adds @adv@ to the total length.
+switchQueue :: IORef Queue -> Int -> Int -> BuildM ()
+switchQueue !qRef !minSize !adv = do
   end <- getCur
-  Queue fptr _ <- io $ readIORef qRef
+  Queue fptr _ total <- io $ readIORef qRef
   let !base = unsafeForeignPtrToPtr fptr
   let !cap = end `minusPtr` base
   newFptr <- if minSize <= cap
     then return fptr
     else io $ mallocForeignPtrBytes minSize
   let !newBase = unsafeForeignPtrToPtr newFptr
-  io $ writeIORef qRef $ Queue newFptr 0
+  io $ writeIORef qRef $ Queue newFptr 0 (total + adv)
```

Comment:
It seems wrong to update the `total` field in both `flushQueue` and `switchQueue`: they are both called in `getBytes_`.

Reply:
One in switchQueue seems to be a mistake. Fixed